service/src/test/java/bio/terra/axonserver/app/controller/GcpResourceControllerTest.java

```diff
@@ -289,4 +302,46 @@ void clusterComponentUrl() throws Exception {
     assertEquals(fakeUrl, url.getUrl());
     Mockito.verify(cluster).getComponentUrl("fakekey");
   }
+
+  @Test
+  void clusterMetadata_get() throws Exception {
+    String operationPath = getClusterOperationPath("metadata");
+
+    GoogleDataprocCluster mockGoogleDataprocCluster = mock(GoogleDataprocCluster.class);
+    Cluster mockCluster = mock(Cluster.class, finalMockSettings);
+    ClusterConfig mockClusterConfig = mock(ClusterConfig.class, finalMockSettings);
+    GceClusterConfig mockGceClusterConfig = mock(GceClusterConfig.class, finalMockSettings);
+    InstanceGroupConfig mockMasterConfig = mock(InstanceGroupConfig.class, finalMockSettings);
+    InstanceGroupConfig mockWorkerConfig = mock(InstanceGroupConfig.class, finalMockSettings);
+
+    doReturn(mockGoogleDataprocCluster)
+        .when(gcpResourceController)
+        .getCluster(workspaceId, resourceId);
+    when(mockGoogleDataprocCluster.getClusterConfig()).thenReturn(mockClusterConfig);
+    when(mockCluster.getConfig()).thenReturn(mockClusterConfig);
+    when(mockClusterConfig.getGceClusterConfig()).thenReturn(mockGceClusterConfig);
+    when(mockClusterConfig.getMasterConfig()).thenReturn(mockMasterConfig);
+    when(mockClusterConfig.getWorkerConfig()).thenReturn(mockWorkerConfig);
+    when(mockClusterConfig.getSecondaryWorkerConfig()).thenReturn(mockWorkerConfig);
+    when(mockMasterConfig.getNumInstances()).thenReturn(1);
+    when(mockWorkerConfig.getNumInstances()).thenReturn(2);
+
+    String serializedGetResponse =
+        mockMvc
+            .perform(
+                get(operationPath)
+                    .contentType("application/json")
+                    .header("Authorization", String.format("bearer %s", fakeToken)))
```

Comment:
minor: you could create a addAuth util function, similar to WSM's mockMvcUtils. might be a used often

Reply:
Thanks, and done. Wasn't aware that we already had these utils in axon server.